aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/Translator.java

```diff
@@ -100,35 +101,38 @@ else if (parameter.applyType().equalsIgnoreCase(ParameterType.Dynamic.toString()
     }
 
 
-    static Set<software.amazon.awssdk.services.rds.model.Tag> translateTagsToSdk(final Map<String, String> tags) {
-        return Optional.ofNullable(tags.entrySet()).orElse(Collections.emptySet())
+    static Map<String, String> translateTagsFromSdk(Collection<Tag> tags) {
+        return Optional.ofNullable(tags).orElse(Collections.emptyList())
                 .stream()
-                .map(tag -> {
-                    return software.amazon.awssdk.services.rds.model.Tag.builder()
-                            .key(tag.getKey())
-                            .value(tag.getValue())
-                            .build();
-                })
-                .collect(Collectors.toSet());
+                .collect(Collectors.toMap(Tag::getKey, Tag::getValue));
     }
 
-    static ResourceModel translateFromDBParameterGroup(
-            final DBParameterGroup dbParameterGroup,
-            final Set<software.amazon.awssdk.services.rds.model.Tag> rdsTags
-    ) {
-        List<Tag> tags = CollectionUtils.isNullOrEmpty(rdsTags) ? null
-                : rdsTags.stream()
-                .map(tag -> Tag.builder()
-                        .key(tag.key())
-                        .value(tag.value())
-                        .build())
+    public static List<software.amazon.awssdk.services.rds.model.Tag> translateTagsToSdk(final Collection<Tag> tags) {
+        return Optional.ofNullable(tags).orElse(Collections.emptyList())
+                .stream()
+                .map(tag -> software.amazon.awssdk.services.rds.model.Tag.builder()
+                        .key(tag.getKey())
+                        .value(tag.getValue())
+                        .build()
+                )
                 .collect(Collectors.toList());
+    }
 
+    static ResourceModel translateFromDBParameterGroup(final DBParameterGroup dbParameterGroup) {
         return ResourceModel.builder()
                 .dBParameterGroupName(dbParameterGroup.dbParameterGroupName())
                 .description(dbParameterGroup.description())
                 .family(dbParameterGroup.dbParameterGroupFamily())
-                .tags(tags)
                 .build();
     }
+
+    static List<Tag> translateTags(final Set<software.amazon.awssdk.services.rds.model.Tag> rdsTags) {
```

Comment:
nit: I believe I agree with the sentiment of using generic containers. `Collection` could make it more universal.

Reply:
Done.